solutions/src/main/java/ru/sberbank/school/task02/ConsoleClient.java

```diff
@@ -0,0 +1,129 @@
+        converter = new Converter(service);
+        cmdParser = new DefaultParser();
+        cmdOptions = new Options();
+        cmdOptions.addOption("o", "operation", true, "ClientOperation");
+        cmdOptions.addOption("s", "symbol", true, "Symbol");
+        cmdOptions.addOption("a", "amount", true, "Amount");
+    }
+
+    @Override
+    public FxResponse fetchResult(FxRequest request) {
+        Symbol symbol = getSymbol(request.getSymbol());
+        ClientOperation operation = getOperation(request.getDirection());
+        BigDecimal amount = getAmount(request.getAmount());
+        if (symbol == null || amount == null || operation == null || amount.compareTo(BigDecimal.ZERO) <= 0) {
+           return makeResponse(request,null);
+        }
+        BigDecimal result = converter.convert(operation,symbol,amount);
+        if (result == null) {
+              return makeResponse(request,null);
+        }
+        return makeResponse(request,result);
+    }
+
+    private FxResponse makeResponse(FxRequest request, BigDecimal result) {
+        LocalDateTime date = LocalDateTime.now();
+        if (result == null) {
+            return new FxResponse(request.getSymbol(),"",
+                                  request.getAmount(),date.format(formatter),
+                                  request.getDirection(),true);
+        }
+
+        BigDecimal amount = new BigDecimal(request.getAmount()).setScale(2, RoundingMode.HALF_UP);
+        result = result.setScale(2,RoundingMode.HALF_UP);
+
+        return new FxResponse(request.getSymbol(),result.toString(),
+                              amount.toString(),date.format(formatter),
+                              request.getDirection(),false);
+    }
+
+    public FxRequest makeRequest(String[] args) {
+        CommandLine cmdLine;
+        try {
+            cmdLine = cmdParser.parse(cmdOptions, args);
+        } catch (ParseException e) {
+            return new FxRequest(null, null, null);
+        }
+
+        String symbol = cmdLine.getOptionValue("s");
+        String operation = cmdLine.getOptionValue("o");
+        String amount = cmdLine.getOptionValue("a");
+
+        if (symbol != null) {
+            symbol = symbol.toUpperCase();
+        }
+        if (operation != null) {
+            operation = operation.toUpperCase();
+        }
+
+        return new FxRequest(symbol, operation, amount);
+    }
+
+    private Symbol getSymbol(String symbol) {
+        if(symbol == null) {
+            return null;
+        }
+        symbol = symbol.replace("/","_");
+        Class symbolClass = Symbol.class;
+        Field[] fields = symbolClass.getFields();
+        for (Field field : fields) {
+            int filedMods = field.getModifiers();
+            if (field.getName().equals(symbol) && Modifier.isStatic(filedMods)) {
+                try {
+                    return (Symbol) field.get(symbolClass);
+                } catch (IllegalAccessException ignored) { }
```

Comment:
Empty catch block.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1395390968>*